scripts/metadata_extractor.py

```
@@ -0,0 +1,253 @@
+                        if first_timestamp is None:
+                            first_timestamp = self._extract_timestamp(line)
+
+                        if is_message:
+                            message_count += 1
+
+                        if text_content:
+                            # Limit text accumulation to prevent memory issues
+                            if len(all_text) < MAX_CONCEPT_MESSAGES:
+                                all_text.append(text_content[:1000])
+
+        except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)